src/Protocol/Protocol_1_8.cpp

```diff
@@ -3310,7 +3310,7 @@ void cProtocol_1_8_0::WriteEntityMetadata(cPacketizer & a_Pkt, const cEntity & a
 			Health: 1 | 3 - (1 - 3) = 5
 			*/
 			auto & Minecart = reinterpret_cast<const cMinecart &>(a_Entity);
-			a_Pkt.WriteBEInt32((((a_Entity.GetMaxHealth() / 2) - (a_Entity.GetHealth() - (a_Entity.GetMaxHealth() / 2))) * Minecart.LastDamage()) * 4);
+			a_Pkt.WriteBEInt32(static_cast<Int32>((((a_Entity.GetMaxHealth() / 2) - (a_Entity.GetHealth() - (a_Entity.GetMaxHealth() / 2))) * Minecart.LastDamage()) * 4));
```

Comment:
I'd guess the line was written like that for clarification.

Reply:
Based on the comment above, I think the intent of the original author was closer to maxHealth - health, so it might be clearest to rewrite that line to be `(a_Entity.GetMaxHealth() - a_Entity.GetHealth()) * Minecart.LastDamage() * 4` and change the comment.

In fact, that's actually how it is implemented in 1.9 and above: https://github.com/cuberite/cuberite/blob/master/src/Protocol/Protocol_1_9.cpp#L3679

Probably not worth changing, since 1.8 is now 3 years old so I assume it's not at the top of our priority list, and it was changed in 1.9 and above.

Reply:
I think I was the original author, yeah it was written for clarification :P